dotcom-rendering/src/web/server/pageTemplate.ts

```diff
@@ -172,20 +193,28 @@ https://workforus.theguardian.com/careers/product-engineering/
                 ${preconnectTags.join('\n')}
                 ${prefetchTags.join('\n')}
 
-                <script type="application/ld+json">
-                    ${JSON.stringify(linkedData)}
-                </script>
+                ${
+					linkedData !== undefined
+						? ` <script type="application/ld+json">
+                    			${JSON.stringify(linkedData)}
+                			</script>`
+						: '<!-- no linked data -->'
+				}
 
                 <!-- TODO make this conditional when we support more content types -->
-                ${ampLink ? `<link rel="amphtml" href="${ampLink}">` : ''}
+                ${
+					ampLink
+						? `<link rel="amphtml" href="${ampLink}">`
+						: '<!-- no Amp link -->'
+				}
```

Comment:
I'm a bit late but I _love_ this change to add comments when something isn't added to the page - can definitely see this coming in handy some day!